server/lib/availabilitySync.ts

```diff
@@ -31,21 +35,65 @@ class AvailabilitySync {
     this.radarrServers = settings.radarr.filter((server) => server.syncEnabled);
     this.sonarrServers = settings.sonarr.filter((server) => server.syncEnabled);
 
+    // Initialize and then ping our plex, radarr, and sonarr client(s) to
+    // check if it they are available before we run the sync
     try {
       await this.initPlexClient();
+    } catch (ex) {
+      logger.error(
+        'Could not connect to your Plex server. Preventing start of the media availability sync.',
+        {
+          errorMessage: ex.message,
+          label: 'AvailabilitySync',
+        }
+      );
+      return;
+    }
 
-      if (!this.plexClient) {
+    if (this.radarrServers.length > 0) {
+      try {
+        await this.initRadarrClient(this.radarrServers);
+      } catch (ex) {
+        logger.error(
+          'Could not connect to your Radarr server(s). Preventing start of the media availability sync.',
+          {
+            errorMessage: ex.message,
+            label: 'AvailabilitySync',
+          }
+        );
         return;
       }
+    } else {
+      logger.info(
+        'Could not find a Radarr server with sync enabled. Skipping Radarr detection.'
+      );
+    }
 
+    if (this.sonarrServers.length > 0) {
+      try {
+        await this.initSonarrClient(this.sonarrServers);
+      } catch (ex) {
+        logger.error(
+          'Could not connect to your Sonarr server(s). Preventing start of the media availability sync.',
+          {
+            errorMessage: ex.message,
+            label: 'AvailabilitySync',
+          }
+        );
+        return;
+      }
+    } else {
+      logger.info(
+        'Could not find a Sonarr server with sync enabled. Skipping Sonarr detection.'
+      );
+    }
```

Comment:
This will initialize your plex, radarr, and sonarr clients to prevent the scan from running if the status check fails.